Wrappers/Python/cil/optimisation/algorithms/FISTA.py

```diff
@@ -62,6 +63,13 @@ class ISTA(Algorithm):
     step_size : positive :obj:`float`, default = None
                 Step size for the gradient step of ISTA.
                 The default :code:`step_size` is :math:`\frac{1}{L}` or 1 if `f=None`.
+   step_size_rule: class with a `get_step_size` method or a function that takes an initialised CIL function as an argument and outputs a step size, default is None
+            This could be a custom `step_size_rule` or one provided in :meth:`~cil.optimisation.utilities.StepSizeMethods`. If None is passed  then the algorithm will use either `ConstantStepSize` or `ArmijioStepSize` depending on if a `step_size` is provided. 
```

Comment:
```suggestion
            This could be a custom `step_size_rule` or one provided in :meth:`~cil.optimisation.utilities.StepSizeMethods`. If None is passed  then the algorithm will use either `ConstantStepSize` or `ArmijoStepSize` depending on if a `step_size` is provided. 
```

Reply:
Thanks Kris - r.e. ` _provable convergence_condition` it raises an error if the step size rule is not constant

Reply:
I will change the naming for Armijo rule

Reply:
> r.e. ` _provable convergence_condition` it raises an error if the step size rule is not constant

sure, but it should also raise an error if `preconditioner` is set, as you don't know what it does, and if you do, knowing what the theoretical limit for the step-size is is going to be hard.

Reply:
Thanks @KrisThielemans - sorry, I misunderstood but that makes sense now